modules/templates/util_json.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package templates
+
+import (
+	"bytes"
+
+	"code.gitea.io/gitea/modules/json"
+)
+
+type JsonUtils struct{} //nolint:revive
+
+func NewJsonUtils() *JsonUtils { //nolint:revive
+	return &JsonUtils{}
```

Comment:
Is the compiler clever and does not allocate a new instance for every call in the template? Maybe there should be a singleton returned by this method.

```go
type JsonUtils struct{}

var instance = &JsonUtils{}

func NewJsonUtils() *JsonUtils {
	return instance
}
```

Reply:
Compiler should be smart enough in this case.

Reply:
`&JsonUtils{}` is quite cheap (comparing to the complex Golang template code) and doesn't seem causing any noticeable problem,  so I didn't make a global variable for it.

And in the future we need to use context function, then we have to make per-request function, global variable doesn't work.

If you prefer to add a global variable cache at the moment (for StringUtils and JsonUtils), it could be fine, but in the future, context functions couldn't do so.

Reply:
Not needed, just curious.

Reply:
Done in 407367c3e3febf0e63bfe63836f75f862dcd2797

Reply:
fyi: The Go compiler uses always the same memory address for empty structs, so the `NewJsonUtils` method may be a noop.

```go
package main

import (
	"fmt"
	_ "unsafe"
)

//go:linkname zerobase runtime.zerobase
var zerobase uintptr

func main() {
	var s struct{}
	var a [42]struct{}

	fmt.Printf("zerobase = %p\n", &zerobase)
	fmt.Printf("       s = %p\n", &s)
	fmt.Printf("       a = %p\n", &a)
}

// zerobase = 0x55c008
//        s = 0x55c008
//        a = 0x55c008
```

https://go.dev/play/p/XMiQhMmPydf